### ISSUE TITLE
When channel is archived, bot throws TypeError in console

### ISSUE THREAD

**Issue by mike503** [slackapi/hubot-slack#183]
Steps to reproduce:

Create channel
/invite bot
Archive channel

Console shows this:

```
[Fri Apr 17 2015 00:06:16 GMT+0000 (UTC)] ERROR TypeError: Cannot read property 'logger' of undefined
  at Channel._onMark (/home/bot/node_modules/hubot-slack/node_modules/slack-client/src/channel.js:192:24)
  at /home/bot/node_modules/hubot-slack/node_modules/slack-client/src/channel.js:2:59
  at IncomingMessage.<anonymous> (/home/bot/node_modules/hubot-slack/node_modules/slack-client/src/client.js:668:22)
  at IncomingMessage.emit (events.js:117:20)
  at _stream_readable.js:944:16
  at process._tickCallback (node.js:448:13)
```

**Comment by DEGoodmanWilson** [comment]
This _should_ be fixed in #309 but I'm going to leave this open so I remember to check.

**Comment by johnagan** [comment]
seems to be fixed. reopen as needed.
